docs/devel/olm-integration.md

```diff
@@ -0,0 +1,151 @@
+
+Last but not least there is a preconditions manifest: if want to test the CSV manifest manually, without
+OperatorSource and Subscription, you can deploy this manifest in order to satisfy all conditions, which are declared
+in the CSV manifest, so that the OLM operator can deploy the KubeVirt operator.  
+
+## Test a new version
+
+Note 1: We use a k8s cluster >= v1.11 for this. You might want to use a OKD cluster with OLM and Marketplace already installed.
+Note 2: You need a Quay.io account
+
+- create manifests with your repository and version info, e.g.:
+  
+  TODO: actually use CSV_VERSION!!!
+  
+  `CSV_VERSION=<csv-version> DOCKER_PREFIX="docker.io/<docker_user>" DOCKER_TAG="<tag>" sh -c 'make generate && make manifests'`
+- verify manifests:
+  `make olm-verify`
+- push images:
+  `DOCKER_PREFIX="index.docker.io/<docker_user>" DOCKER_TAG="<tag>" make bazel-push-images`
+- push the operator bundle:
+  `CSV_VERSION=<csv-version> QUAY_USER=<username> QUAY_PASSWORD=<password> make olm-push`
+  Note: you need to update the CSV version (and so run `make manifests`) on every push! (or maybe delete an old version before pushing again?)
+  
+- install OLM and Marketplace (see below)
+
+- install KubeVirt OperatorSource:
+  `cd _out/manifests/release/olm`
+  `kubectl apply -f kubevirt-operatorsource.yaml`
+- check that a CatalogSourceConfig and a CatalogSource were created in the marketplace namespace
+- WORKAROUND: the OKD console only shows operators from CatalogSources in the `olm` namespace. In order to get it there,
+  you need to edit the CatalogSourceConfig and change the targetNamespace from `marketplace` to `olm`. The new
+  CatalogSource should be created automatically.
+- create the kubevirt namespace:
+  `kubectl create ns kubevirt`
+- install the OperatorGroup for the new namespace:
+  `kubectl apply -f operatorgroup.yaml`
+- create a Subscription:
+  `kubectl apply -f kubevirt-subsription.yaml`
+- check that a InstallPlan was created
+- check that the KubeVirt operator was installed
+- install a KubeVirt CR
+
+Bonus: install the OKD Console:
+
+- we need cluster-admin permissions for the kube-system:default account:
+  `kubectl create clusterrolebinding defaultadmin --clusterrole cluster-admin --serviceaccount kube-system:default`
+- in the OLM repository, run `./scripts/run_console_local.sh`
+- open `localhost:9000` in a browser
+
+## Release a new version
+
+Travis cares for this on every release.
+
+## Installing OLM on Kubernetes
+
+- clone github.com/operator-framework/operator-lifecycle-manager
+- `cd deploy/upstream/quickstart`
+- `kubectl apply -f olm.yaml`
+- if you get an error, try again, CRDs might have been too slow
+
+## Installing Marketplace on Kubernetes
+
+- clone github.com/operator-framework/operator-marketplace
+- `cd deploy/upstream/manifests`
+- `kubectl apply -f upstream/`
+- if you get an error about rolebinding, repeat with `--validate=false`
+
+## Sources
```

Comment:
Change plain urls to hyperlinks, please, everywhere.

Reply:
done